**What dataset are we working with?**

https://www.kaggle.com/datasets/mathurinache/citation-network-dataset/data?select=dblp.v12.json

How was the dataset collected?

For what purpose was the dataset collected?

## RUN clean_data.py OR GET data.csv FROM GOOGLE DRIVE BEFORE RUNNING THIS NOTEBOOK!

In [2]:
!pip freeze

annotated-types==0.6.0
anyio==4.3.0
argcomplete==3.1.1
argon2-cffi==23.1.0
argon2-cffi-bindings==21.2.0
asciinema==2.4.0
asttokens==2.4.1
async-lru==2.0.4
attrs==23.2.0
autocommand==2.2.2
Babel==2.14.0
beautifulsoup4==4.12.3
bleach==6.1.0
btrfsutil==6.8
certifi==2024.2.2
cffi==1.16.0
charset-normalizer==3.3.2
click==8.1.7
colorama==0.4.6
comm==0.2.2
cryptography==42.0.5
dbus-python==1.3.2
decorator==5.1.1
defusedxml==0.7.1
discid==1.2.0
executing==2.0.0
fastjsonschema==2.19.1
h11==0.14.0
httpcore==1.0.2
httpx==0.26.0
idna==3.6
importlib_metadata==5.1.0
inflect==7.2.0
ipykernel==6.29.4
ipython==8.22.2
jaraco.context==4.3.0
jaraco.functools==4.0.0
jaraco.text==3.12.0
jedi==0.19.1
Jinja2==3.1.3
json5==0.9.25
jsonschema==4.21.1
jsonschema-specifications==2023.12.1
jupyter-events==0.10.0
jupyter_client==8.6.1
jupyter_core==5.7.2
jupyter_server==2.14.0
jupyterlab==4.1.6
jupyterlab_pygments==0.3.0
jupyterlab_server==2.26.0
markdown-it-py==3.0.0
MarkupSafe==2.1.5
matplotlib-inline==0.1.6
mdurl

In [1]:
import pandas as pd
import numpy as np

citations_df = pd.read_csv('data.csv')

citations_df.head()

ModuleNotFoundError: No module named 'pandas'

In [ ]:
citations_df.columns

Index(['ID', 'Title', 'Year', 'Citations', 'Document Type', 'Authors',
       'Venue'],
      dtype='object')

In [ ]:
# Trim out the rows with n_citations = 0
print(citations_df.shape)

There are 4894081 rows of data.

In [ ]:
# These columns are Series data types
# print(citations_df.Year.min())
# print(citations_df.Year.max())

# Look for papers whose year isn't 0
years_citations_df = citations_df[citations_df["Year"] != 0]
print(years_citations_df.shape)
print(years_citations_df.Year.min())
print(years_citations_df.Year.max())

There are 9 papers whose Year is 0.

In [ ]:
# Papers written between 1800 (inclusive) and 1899 (inclusive)
print(years_citations_df[(years_citations_df["Year"] >= 1800) & (years_citations_df["Year"] < 1900)])

In [ ]:
# Papers written between 1900 (inclusive) and 1999 (inclusive)
print(years_citations_df[(years_citations_df["Year"] >= 1900) & (years_citations_df["Year"] < 1999)])

In [ ]:
# Papers written past 2000 (inclusive)
print(years_citations_df[(years_citations_df["Year"] >= 2000)])

In [ ]:
# testingData = {
#     'Name': ['John', 'Doe', 'Alice', 'Bob', 'Chris'],
#     'Age': [25, 30, 22, 28, 35],
#     'Salary': [50000, 60000, None, 75000, 90000],
#     'Experience': [2, 5, 1, None, 10]
# }

# testing_df = pd.DataFrame(testingData)

# print(testing_df["Salary"].isna())
# print(testing_df["Salary"].isna().sum())

0    False
1    False
2     True
3    False
4    False
Name: Salary, dtype: bool
1


In [ ]:
print(citations_df["ID"].isna().sum())
print(citations_df["Title"].isna().sum())

print(len(citations_df.Title.unique()))
print("There are " + str(citations_df.shape[0] - len(citations_df.Title.unique())) + " papers that share the same title" )

0
0
4696970
There are 197111 papers that share the same title


Of the 4894081 papers in the dataset, 197111 papers share the same title.

In [ ]:
print(citations_df.Citations.min())
print(citations_df.Citations.max())

print(citations_df[citations_df["Citations"] >= 48327])
print(citations_df[citations_df["Citations"] == 0])

0
48327
                 ID                                     Title  Year  \
4696136  2041404167  The Mathematical Theory of Communication  1949   

         Citations Document Type        Authors Venue  
4696136      48327          Book  C. E. Shannon        
                ID                                              Title  Year  \
4             5411  COMPARING GNG3D AND QUADRIC ERROR METRICS METH...  2009   
6             6522  Improved Secret Image Sharing Method By Encodi...  2011   
7             6762  A Self-Stabilizing Algorithm for Finding the C...  2003   
8             8373  Formal agent-oriented ubiquitous computing: a ...  2012   
9             8763  Fur Visualisation for Computer Game Engines an...  2014   
...            ...                                                ...   ...   
4894070  999760546  Trust, Privacy and Security in Digital Busines...  2011   
4894075  998513720       Communication: Words and Conceptual Systems.  2015   
4894077   99884553   Blind